feat(core): Introduce @sentry/expo-upload-sourcemaps package (#6027)
getsentry/sentry-react-native: CHANGELOG.md

### Features

> - Add new `@sentry/expo-upload-sourcemaps` package for uploading JavaScript bundles and source maps from Expo builds to Sentry. Equivalent to the existing `sentry-expo-upload-sourcemaps` bin bundled with `@sentry/react-native`, which continues to work unchanged. ([#6027](https://github.com/getsentry/sentry-react-native/pull/6027))

Author: ```suggestion
- Add new `@sentry/expo-upload-sourcemaps` package for uploading JavaScript bundles and source maps from Expo builds to Sentry ([#6027](https://github.com/getsentry/sentry-react-native/pull/6027))
  - The existing `sentry-expo-upload-sourcemaps` bin bundled with `@sentry/react-native` is deprecated
```
